cypress/integration/tip.test.ts

```diff
@@ -1,80 +1,98 @@
-      expect(response.body).to.have.property('minimalFee') // true
-      expect(response.body).to.have.property('feeRatio') // true
-      expect(response.body).to.have.property('expirationDate') // true
-    })
+const FEE_QUERY = `https://protocol-rinkeby.dev.gnosisdev.com/api/v1/tokens/${WETH[4].address}/fee`
+const SWAP_URL = '/swap'
+
+describe('[EXACT-IN::SELL] :: ETH/DAI', () => {
+  before(() => {
+    cy.visit(SWAP_URL)
   })
 
-  it('Swap ETH for DAI and applies fee', () => {
+  it('[UI] Selects DAI token', () => {
     // Open currency selector
     cy.get('#swap-currency-output .open-currency-select-button').click()
-    cy.get(`.token-item-${DAI}`).should(`be.visible`)
+    cy.get(`.token-item-${OUTPUT_TOKEN.address}`).should(`be.visible`)
     // select DAI
-    cy.get(`.token-item-${DAI}`).click({ force: true })
+    cy.get(`.token-item-${OUTPUT_TOKEN.address}`).click({ force: true })
     cy.get('#swap-currency-input .token-amount-input').should('be.visible')
     // input the cached input amount (above)
-    cy.get('#swap-currency-input .token-amount-input').type(INPUT_AMOUNT, { force: true, delay: 200 })
+    cy.get('#swap-currency-input .token-amount-input').type(TYPED_AMOUNT, { force: true, delay: 200 })
     cy.get('#swap-currency-output .token-amount-input').should('not.equal', '')
+  })
+
+  context('[Trade + Fee Logic Tests] :: ETH/DAI', () => {
+    beforeEach(() => {
+      // set aliases in context to reuse
+      // start fee request and alias as @fee
+      cy.request(FEE_QUERY)
+        .its('body')
+        .as('fee')
+
+      // Trade object and alias as @trade
+      cy.window()
+        .as('win')
+        .its('trade')
+        .as('trade')
+
+      cy.get('#swap-currency-output .token-amount-input').as('output')
+
+      // Clear the storage (old redux state etc)
+      cy.get<Window>('@win').then(window => window.localStorage.clear())
+    })
+
+    it('[TRADE] Trade properties are correct', () => {
+      cy.get<Trade>('@trade').should((trade: Trade) => {
+        // Trade should have input/out amounts and executionPrice
+        expect(trade).to.have.property('inputAmount')
+        expect(trade).to.have.property('outputAmount')
+        expect(trade).to.have.property('executionPrice')
+      })
+
+      cy.get<Trade>('@trade')
+        .its('inputAmount')
+        .should(inputAmount => {
+          expect(inputAmount.toExact()).to.equal(TYPED_AMOUNT)
+        })
+
+      cy.get<Trade>('@trade').should(trade => {
+        const { executionPrice, inputAmount } = trade
+        // OUTPUT / PRICE = INPUT (as string)
+        const calculatedOutputAmount = inputAmount.multiply(executionPrice).toSignificant(inputAmount.currency.decimals)
+        expect(calculatedOutputAmount).to.not.be.null
+        // cy.get('@output').should('equal', calculatedOutputAmount)
+      })
+    })
+
+    it('[FEE] NO FEE', () => {
+      cy.get<FeeInformation>('@fee').should(fee => {
```

Comment:
I don't get this test either, u are calling the API and expecting to return 0 fee?
what if it changes?

Reply:
I have the same question, but I guess this will come later?